### DISCUSSION TITLE
[Thanos] Consistent Hashing in Thanos Receive Hashrings

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#613] [Google Summer of Code 2022]
The current implementation of the Thanos Receive uses non-consistent hashing to distribute metrics across distributed ingesting replicas. The scalability that this enables can be improved. Because the hash is not consistent, every change in number or replicas considered for metric distribution can cause larger memory utilization for short periods. In the past, we tried to mitigate this problem by flushing the replica content to object storage which caused delays in scale-out and down. Switching to consistent hash implementation like a hash ring would allow us to mitigate this issue without delaying the scaling process. This should improve the life of Thanos receive users by enabling easier auto-scaling capabilities for production systems that have to react to different metric consumption characteristics. Join us in the effort of moving the hashing method to a consistent one!

Mentor(s): Lucas Servén Marín (@squat), Prem Saraswat (@onprem), Matej Gera (@matej-g)

https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#consistent-hashing-in-thanos-receive-hashrings

**Comment by rvacaru** [comment]
Hi Thanos people,
I'm Razvan Vacaru, a data engineer based in Amsterdam. I'm looking into the GSoC 2022 ideas and this is one (but also the other one about [compaction/downsampling](https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#compactiondownsampling-build-chunks-from-object-storage-without-touching-disks)) that I think I'll like and I'll be able to help with.

In short my CV is:
- Bachelor's degree in CS Engineering (Java, C, Javascript, Linux sysadmin)
- Worked ~6 years mostly as a Java/Spring developer (4 of the 6 years in AWS)
- Working for ~ 2 years as a Data engineer with python, spark, terraform, datadog, ...
- Knowledge of distributed systems like cassandra, spark and hadoop

Check the full CV in LinkedIn: https://www.linkedin.com/in/razvan-vacaru-17787b5a/
From time to time I also do small contributions to OS projects like akka or dbt: https://github.com/rvacaru

My goal is to increase my knowledge of the K8s ecosystem, monitoring and Golang.
Do you think I would be a good fit for this project?
